commercetools-java-client-core/src/main/java/io/sphere/sdk/client/SolutionInfoService.java

```diff
@@ -21,9 +20,20 @@ public static synchronized SolutionInfoService getInstance() {
     }
 
     public List<SolutionInfo> getSolutionInfos() {
-        final List<SolutionInfo> solutions = new LinkedList<>();
-        loader.iterator()
-                .forEachRemaining(solution -> solutions.add(solution));
+        List<SolutionInfo> solutions  ;
+        ServiceLoader<SolutionInfo> loader;
+        // workaroud for play framework, since the overloaded version of ServiceLoader seems to cause some errors there, here we try bot until we get a working classloader
+        try{
+            loader = ServiceLoader.load(SolutionInfo.class);
+            solutions = new ArrayList<>();
+            loader.forEach(solutions::add);
+
+        }catch(Throwable e){
+            loader = ServiceLoader.load(SolutionInfo.class,SolutionInfo.class.getClassLoader());
+            solutions = new ArrayList<>();
+            loader.forEach(solutions::add);
```

Comment:
And we could have moved this `forEach` out of the try clause.

Reply:
Actually its during the insertion where the cast issue appears

Reply:
I see 😄 and thanks for the explanation. I wasn't aware that the `load` method returns an `Iterator` 😉